scripts/pin_the_slow_way.py

```diff
@@ -25,38 +25,90 @@
+# https://abi-laboratory.pro/tracker/timeline/freetype/
+# https://abi-laboratory.pro/tracker/timeline/openssl/
+# https://abi-laboratory.pro/tracker/timeline/sqlite/
+# https://abi-laboratory.pro/tracker/timeline/qt/
+# https://abi-laboratory.pro/tracker/timeline/giflib/
+# http://upstream.rosalinux.ru/versions/libgeos_c.html
+# http://upstream.rosalinux.ru/versions/libgeos.html
+# http://upstream.rosalinux.ru/versions/netcdf.html
+# http://upstream.rosalinux.ru/versions/readline.html
+# http://upstream.rosalinux.ru/versions/tk.html
+# http://upstream.rosalinux.ru/versions/libsox.html
+# http://upstream.rosalinux.ru/versions/zlib.html
+# http://upstream.rosalinux.ru/versions/ncurses.html
+# http://upstream.rosalinux.ru/versions/blitz.html
+
+build_pin = {
           'boost': 'boost 1.61.*',
           'bzip2': 'bzip2 1.0.*',
           'ffmpeg': 'ffmpeg 2.8.*',
           'fontconfig': 'fontconfig 2.11.*',
           'freetype': 'freetype 2.6.*',
           'geos': 'geos 3.4.*',
-          'giflib': 'giflib 5.1.*',
+          'giflib': 'giflib 5.1.2',
           'hdf5': 'hdf5 1.8.17|1.8.17.*',
           'icu': 'icu 56.*',
           'jpeg': 'jpeg 9*',
           'libblitz': 'libblitz 0.10|0.10.*',
           'libmatio': 'libmatio 1.5.*',
           'libnetcdf': 'libnetcdf 4.4.*',
-          'libpng': 'libpng >=1.6.23,<1.7',
+          'libpng': 'libpng 1.6.21',
           'libsvm': 'libsvm 3.21|3.21.*',
           'libtiff': 'libtiff 4.0.*',
           'ncurses': 'ncurses 5.9*',
           'netcdf-cxx4': 'netcdf-cxx4 4.3.*',
           'netcdf-fortran': 'netcdf-fortran 4.4.*',
           'openblas': 'openblas 0.2.18|0.2.18.*',
-          'openssl': 'openssl 1.0.*',
+          'openssl': 'openssl 1.0.2h',
           'proj.4': 'proj.4 4.9.3',
-          'pyqt': 'pyqt 4.11.*',
-          'qt': 'qt 4.8.*',
+          'pyqt': 'pyqt 4.11.4',
+          'qt': 'qt 4.8.5',
           'readline': 'readline 6.2*',
           'sox': 'sox 14.4.2',
           'sqlite': 'sqlite 3.13.*',
           'tk': 'tk 8.5.*',
           'vlfeat': 'vlfeat 0.9.20',
           'xz': 'xz 5.2.*',
-          'zlib': 'zlib 1.2.*',
+          'zlib': 'zlib 1.2.8',
+        }
+
+run_pin = {
+          'boost': 'boost 1.61.*',
+          'bzip2': 'bzip2 >=1.0,<2',
+          'ffmpeg': 'ffmpeg 2.8.*',
+          'fontconfig': 'fontconfig >=2.11,<3',
+          'freetype': 'freetype 2.6.*',
+          'geos': 'geos 3.4.*',
+          'giflib': 'giflib >=5.1.2,<6',
+          'hdf5': 'hdf5 1.8.17|1.8.17.*',
+          'icu': 'icu 56.*',
+          'jpeg': 'jpeg 9*',
+          'libblitz': 'libblitz 0.10|0.10.*',
+          'libmatio': 'libmatio 1.5.*',
+          'libnetcdf': 'libnetcdf >=4.4,<5',
+          'libpng': 'libpng >=1.6.21,<1.7',
+          'libsvm': 'libsvm 3.21|3.21.*',
+          'libtiff': 'libtiff >=4.0,<5',
+          'ncurses': 'ncurses >=5.9,<6',
+          'openblas': 'openblas 0.2.18|0.2.18.*',
+          'openssl': 'openssl >=1.0.2h,<1.1',
```

Comment:
I see based on this: https://www.openssl.org/docs/faq.html#MISC8 you want to pin something like 'openssl >=1.0.2[any letter],<1.1' but how do you do that? This: 'openssl >=1.0.2*,<1.1' is not good.